.github/workflows/ci.yml

```diff
@@ -0,0 +1,38 @@
+name: test
+on:
+  pull_request:
+    branches: [ main ]
+jobs:
+  codegen:
+    name: Codegen
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Restore go build cache #This action allows caching dependencies and build outputs to improve workflow execution time 
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.mod') }}
+      - name: Setup Go Environment #https://github.com/marketplace/actions/setup-go-environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.17.1' # The Go version to download (if necessary) and use
+      - name: Install mockgen
+        run: |
+          MOCKGEN_ZIP=mock_1.6.0_linux_amd64.tar.gz
+          curl -OL https://github.com/golang/mock/releases/download/v1.6.0/$MOCKGEN_ZIP
+          sudo tar xvzf $MOCKGEN_ZIP --strip-components 1 -C /usr/local
+          sudo chmod +x /usr/local/mockgen
+          rm -f $MOCKGEN_ZIP
+          echo /usr/local/ >> $GITHUB_PATH
+      - name: Get dependencies
+        run: go mod download
+      - name: Generate mocks
+        run:  ./generate_mocks.sh
+      - name: Ensure nothing changed
+        run: git diff --exit-code
```

Comment:
Is this all that is required? I think we need to actually check that the exit code is 0. 

```
go-tfe % git diff --exit-code
go-tfe % echo $?
0
```